packages/datadog-instrumentations/src/graphql.js

```diff
@@ -271,27 +262,15 @@ function assertField (context, info, args) {
 
   if (!field) {
     const parent = getParentField(context, path)
-
-    // we want to spawn the new span off of the parent, not a new async resource
-    parent.asyncResource.runInAsyncScope(() => {
-      /* this child resource will run a branched scope off of the parent resource, which
-      accesses the parent span from the storage unit in its own scope */
-      const childResource = new AsyncResource('bound-anonymous-fn')
-
-      childResource.runInAsyncScope(() => {
-        startResolveCh.publish({
-          info,
-          context,
-          args
-        })
-      })
-
-      field = fields[pathString] = {
-        parent,
-        asyncResource: childResource,
-        error: null
-      }
-    })
+    // we need to pass the parent span to the field if it exists for correct span parenting
```

Comment:
This should be handled on the plugin side. Instrumentation should never know about spans.

Reply:
Basically, the plugin should be tracking the hierarchy of resolvers through the context and any information on the context so that it can do the correct parentage. It should really always have been the plugin handling this, but with `AsyncResource` that was not really possible.